src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(num) {
+    if (typeof num === 'number') {
+      sum += num;
+
+      return adder;
+    }
+
+    const result = sum;
+
+    sum = 0;
+
+    return result;
```

Comment:
The current implementation of the `adder` function will not work correctly if a falsy value that is not `undefined` is passed, such as `0`. This will incorrectly reset the sum. To fix this, explicitly check for `undefined` instead of a falsy value.